sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -549,7 +562,7 @@ public final class StorageBlobClient: PipelineClient {
         ///   - name: The name of the blob.
         ///   - container: The name of the container.
         public static func localUrl(
-            inDirectory directoryUrl: URL = cacheDir,
+            inDirectory directoryUrl: URL = url(for: .cachesDirectory),
```

Comment:
Instead of us arbitrarily choosing a default, what if you configured the default? `BlobStorageClient.PathHelper.defaultDirectory = .foo`. That way if I want something other than cachesDirectory I don't have to manually include it with every call?

Reply:
I started working on this, and ended up with a fairly different idea which I think improves ergonomics and correctness quite a bit. Please check out the most recent commit, and especially the full commit message.